proto/tpm_enrollz.proto

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" B1IS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+syntax = "proto3";
+
+package openconfig.attestz.proto;
+
+import "github.com/openconfig/attestz/proto/common_definitions.proto";
+
+option go_package = "github.com/openconfig/attestz";
+
+message GetIakCertRequest {
+  // Switch control card selected identifier.
+  ControlCardSelection control_card_selection = 1;
+}
+
+message GetIakCertResponse {
+  // Vendor identity fields of the selected control card.
+  ControlCardVendorId control_card_id = 1;
+
+  // PEM-encoded initial attestation key certificate signed by the switch
+  // vendor CA.
+  string iak_cert = 2;
+
+  // PEM-encoded initial DevID certificate signed by the switch vendor CA.
+  // This field should only be populated for standby/secondary control card.
+  string idevid_cert = 3;
+}
+
+message StoreOIakCertRequest {
+  // Switch control card selected identifier.
+  ControlCardSelection control_card_selection = 1;
+
+  // PEM-encoded owner initial attestation key certificate signed by the
+  // switch owner/administrator CA.
+  string oiak_cert = 2;
+
+  // PEM-encoded owner initial DevID certificate signed by the
+  // switch owner/administrator CA.
+  string oidevid_cert = 3;
+}
+
+message StoreOIakCertResponse {
+}
+
+service TpmEnroller {
```

Comment:
I am open to renaming it. Do you suggest calling something like `EnrollZ` or `EnrollzService`? Any other ideas?

Yes, TPM enrollment and attestation should be separate services or at least we should not force them to be part of the same service (in fact, at Google they are two separate services). 

In TPM 2.0 world with the introduction of IAK/LAK and IDevID/LDevID capabilities, one does not even have to have "TPM enrollment" on the owner side and can simply rely on IAK and IDevID as opposed to oIAK and oIDevID. At Google we do TPM enrollment, so that we have full control over attestation cert revocation/lifetime/structure/rotation and do not depend on the vendor CA. 
Secondly, semantically speaking, TPM enrollment is about verifying device identity whereas attestation is about verifying device boot state, so again there can be a good argument made to separate those functionalities into separate services and scope of team ownership.

Reply:
TPMEnrollzService to match the other PR?